admin_app/src/app/user-management/page.tsx

```diff
@@ -0,0 +1,199 @@
+"use client";
+import React from "react";
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Grid,
+  List,
+  Paper,
+  Typography,
+} from "@mui/material";
+import { UserCardNew } from "./components/UserCard";
+import { editUser, getUserList, resetPassword, UserBodyPassword } from "./api";
+import { useAuth } from "@/utils/auth";
+import { CreateUserModal, EditUserModal } from "./components/UserCreateModal";
+import { ConfirmationModal } from "./components/ConfirmationModal";
+import { createUser, UserBody } from "./api";
+import { UserResetModal } from "./components/UserResetModal";
+import { appColors } from "@/utils";
+import { Layout } from "@/components/Layout";
+
+const UserManagement: React.FC = () => {
+  const { token, role } = useAuth();
+  const [users, setUsers] = React.useState<UserBody[]>([]);
+  const [showCreateModal, setShowCreateModal] = React.useState(false);
+  const [showEditModal, setShowEditModal] = React.useState(false);
+  const [showUserResetModal, setShowUserResetModal] = React.useState(false);
+  const [currentUser, setCurrentUser] = React.useState<UserBody | null>(null);
+  const [loading, setLoading] = React.useState(true);
+  const [recoveryCodes, setRecoveryCodes] = React.useState<string[]>([]);
+  const [showConfirmationModal, setShowConfirmationModal] = React.useState(false);
+  const [hoveredIndex, setHoveredIndex] = React.useState<number>(-1);
+  React.useEffect(() => {
+    getUserList(token!).then((data: UserBody[]) => {
+      const sortedData = data.sort((a: UserBody, b: UserBody) =>
+        a.username.localeCompare(b.username),
+      );
+      setLoading(false);
+      setUsers(sortedData);
+    });
+  }, [loading]);
+  React.useEffect(() => {
+    if (recoveryCodes.length > 0) {
+      setShowConfirmationModal(true);
+    } else {
+      setShowConfirmationModal(false);
+    }
+  }, [recoveryCodes]);
+  const handleRegisterModalContinue = (newRecoveryCodes: string[]) => {
+    setRecoveryCodes(newRecoveryCodes);
+    setLoading(true);
+    setShowCreateModal(false);
+  };
+  const handleEditModalContinue = (newRecoveryCodes: string[]) => {
+    setLoading(true);
+    setShowEditModal(false);
+  };
+
+  const handleResetPassword = (user: UserBody) => {
+    setCurrentUser(user);
+    setShowUserResetModal(true);
+  };
+
+  const handleEditUser = (user: UserBody) => {
+    setCurrentUser(user);
+    setShowEditModal(true);
+  };
+
+  if (role !== "admin") {
+    return (
+      <Box display="flex" justifyContent="center" alignItems="center" height="100vh">
+        <Typography variant="h4" color="error">
+          [403] Access Denied
+        </Typography>
+      </Box>
+    );
+  }
+
+  return (
```

Comment:
Can I suggest improving visuals in either this PR or the next? It looks a bit messy

<img width="1398" alt="image" src="https://github.com/user-attachments/assets/abb9c4d0-c9dd-4428-8813-e37d7b6fa49e">

Can we align title like we do with urgency detection, fix font sizes (probably should be smaller), and reduce the spacing between fields here.



<img width="424" alt="image" src="https://github.com/user-attachments/assets/10bc28c2-bc0f-4434-b000-93b4a416717c">

Reply:
I can make the changes you asking here, but I suggest we do more extensive improvements in the next PR.